File: update_data.dart
// Random Dart class generated on 2025-06-26

class CheckItem {
  String _name;
  int _value;

  CheckItem(this._name, this._value);

  String get name => _name;
  int get value => _value;

  set name(String newName) {
    _name = newName;
  }

  set value(int newValue) {
    _value = newValue;
  }

  Map<String, dynamic> toJson() {
    return {
      'name': _name,
      'value': _value,
    };
  }
}

File: update_data.c
// Random C struct generated on 2025-09-07
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    char name[100];
    int value;
    int active;
    long timestamp;
} GetItem;

GetItem* create_getitem(const char* name, int value) {
    GetItem* obj = malloc(sizeof(GetItem));
    if (obj == NULL) return NULL;
    
    strncpy(obj->name, name, sizeof(obj->name) - 1);
    obj->name[sizeof(obj->name) - 1] = '\0';
    obj->value = value;
    obj->active = 1;
    obj->timestamp = 1423630974;
    
    return obj;
}

void update_getitem_value(GetItem* obj, int new_value) {
    if (obj != NULL) {
        obj->value = new_value;
    }
}

void print_getitem(const GetItem* obj) {
    if (obj != NULL) {
        printf("Name: %s, Value: %d, Active: %d\n", 
               obj->name, obj->value, obj->active);
    }
}

void free_getitem(GetItem* obj) {
    if (obj != NULL) {
        free(obj);
    }
}

int main() {
    GetItem* obj = create_getitem("sample", 58);
    print_getitem(obj);
    free_getitem(obj);
    
    return 0;
}
